src/test_workflow/test_args.py

```diff
@@ -38,21 +37,13 @@ def __init__(self):
         self.component = args.component
         self.keep = args.keep
         self.logging_level = args.logging_level
-        self.test_manifest_path = self._validate_path(args.test_manifest_path)
-
-        self.opensearch_path = self._validate_paths(args.paths, "opensearch")
-        self.opensearch_dashboards_path = self._validate_paths(args.paths, "opensearch-dashboards")
-
-    def _validate_paths(self, args_paths: dict, name: str) -> str:
-        path = "."
-
-        if args_paths and (name in args_paths):
-            path = args_paths[name]
+        self.test_manifest = TestManifest.from_path(args.test_manifest_path)
 
-        return self._validate_path(path)
+        cwd = os.getcwd()
+        self.opensearch_path = args.paths.get("opensearch", cwd) if args.paths else cwd
 
-    def _validate_path(self, path: str) -> str:
-        return path if validators.url(path) else os.path.realpath(path)
+        if self.test_manifest.name == "OpenSearch Dashboards":
```

Comment:
Why do you need the `if`? Just assign it if it's passed in. Or better, assign `self.paths` and use `self.paths.get("...")` as needed.

Reply:
E.g 
1) `./build.sh manifests/1.3.0/opensearch-1.3.0.yml`

2) `./build.sh manifests/1.3.0/opensearch-dashboards-1.3.0.yml`

Both will set the default value for OpenSearch. But only 2) will need to set a default value for Dashboards. I think this is also what Peter means in this comment https://github.com/opensearch-project/opensearch-build/pull/1263#discussion_r765899016

Reply:
First, default paths to `{}` and assign paths to `self`. This way it's always a dictionary and you don't have to do `if args.paths` ever.

```
self.paths = args.paths
```

Now, everywhere you need a dashboards path you could do `self.paths.get("opensearch-dashboards", cwd)`, or if you must, declare it as a property.

```
@property
def opensearch_dashboards_path(self)
    self.paths.get("opensearch-dashboards", cwd)
```

Does this remove the `if` or am I missing something?

Reply:
Great suggestion! Originally my mind was limited to solve this problem within the scope of `TestArgs` class. But your suggestion opened my mind to let the caller (who intrinsically knows the situation better than `TestArgs` ) decide the parameters to use. 

Just updated the code and it is much cleaner and elegant!